src/Main.cpp

```diff
@@ -1,10 +1,467 @@
+    content.append(decoder.getSessionIdFromList());
+    message["arguments"] = content;
+    message["invocationId"] = "0";
+    message["streamIds"] = Json::arrayValue;
+    message["target"] = "leave_session";
+    message["type"] = 1;
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void pong(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder,
+    string const& local_peer_id
+)
+{
+    Json::Value message;
+    message["protocol"] = "one-to-one";
+    message["to"] = decoder.getFrom();
+    message["action"] = "pong";
+    message["data"]["from"] = local_peer_id;
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void offerAnswerRustyMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["target"] = decoder.getTo();
+    message["caller"] = decoder.getFrom();
+    message["sdp_message"]["type"] = decoder.getActionType();
+    message["sdp_message"]["sdp"] = decoder.getDescription();
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void candidateRustyMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["target"] = decoder.getTo();
+    message["candidate"] = decoder.getCandidate();
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void offerAnswerSignalMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["protocol"] = "one-to-one";
+    message["to"] = decoder.getTo();
+    message["action"] = decoder.getActionType();
+    message["data"]["from"] = decoder.getFrom();
+    message["data"]["description"] = decoder.getDescription();
+
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void candidateSignalMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["protocol"] = "one-to-one";
+    message["to"] = decoder.getTo();
+    message["action"] = decoder.getActionType();
+    message["data"]["from"] = decoder.getFrom();
+    message["data"]["candidate"] = decoder.getDescription();
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+bool initialHandShakeFinalized(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder,
+    string& local_peer_id,
+    bool& client_id_checked
```

Comment:
Right

So, this function returns true when the client id is checked ... Just use the return value to set `client_id_checked` in the caller.

Reminder: https://docs.google.com/document/d/16zBGRa58zu7wgRBcGb6IVVQtCfiDqDDhu4vjUVnVSu0/edit#heading=h.tu1h6jll3wc